Source/Core/VideoCommon/LightingShaderGen.cpp

```diff
@@ -83,94 +104,135 @@ static void GenerateLightShader(ShaderCode& object, const LightingUidData& uid_d
 void GenerateLightingShaderCode(ShaderCode& object, const LightingUidData& uid_data,
                                 std::string_view in_color_name, std::string_view dest)
 {
-  for (u32 j = 0; j < NUM_XF_COLOR_CHANNELS; j++)
+  for (u32 chan = 0; chan < NUM_XF_COLOR_CHANNELS; chan++)
   {
-    object.Write("{{\n");
-
-    const bool colormatsource = !!(uid_data.matsource & (1 << j));
-    if (colormatsource)  // from vertex
-      object.Write("int4 mat = int4(round({}{} * 255.0));\n", in_color_name, j);
-    else  // from color
-      object.Write("int4 mat = {}[{}];\n", I_MATERIALS, j + 2);
-
-    if ((uid_data.enablelighting & (1 << j)) != 0)
+    const u32 chan_a = chan + NUM_XF_COLOR_CHANNELS;  // Data for alpha is stored after all colors
```

Comment:
It's likely a bit nicer from a readability standpoint to put the comment above the line rather than off to the side in this instance